core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -500,6 +499,38 @@ public final boolean isSystemCheck() {
 
     }
 
+    /**
+     * Checks if the serial collector is in use. This collector is single-threaded and devastating
+     * for performance and should not be used for a server application like Elasticsearch.
+     */
+    static class UseSerialGCCheck implements BootstrapCheck.Check {
+
+        @Override
+        public boolean check() {
+            return getUseSerialGC().equals("true");
```

Comment:
I don't think the bootstrap check should fail if it returns a funny string, I do think we can consider such things like logging in `JvmInfo` though; I'll take a look at that.